pymodbus/transaction.py

```diff
@@ -379,17 +378,6 @@ def _recv(self, expected_response_length, full):  # noqa: C901
                         )
                         length = struct.unpack(">H", read_min[4:6])[0] - 1
                         expected_response_length = h_size + length
-                    elif expected_response_length is None and isinstance(
-                        self.client.framer, ModbusRtuFramer
-                    ):
-                        with suppress(
-                            IndexError  # response length indeterminate with available bytes
-                        ):
-                            expected_response_length = (
-                                self.client.framer.get_expected_response_length(
```

Comment:
Actually I think this needs to be changed to calling these 2 methods:
```
pdu_class = self.decoder.lookupPduClass(func_code)
         return pdu_class.calculateRtuFrameSize(data)
```
I am quite sure I should not have removed it totally.